common/src/main/java/com/wynntils/screens/maps/IconFilterScreen.java

```diff
@@ -0,0 +1,346 @@
+    }
+
+    private void scroll(int delta) {
+        int maxValue = getMaxScrollOffset();
+
+        // Calculate how many rows should be scrolled past
+        scrollOffset = MathUtils.clamp(scrollOffset + delta, 0, maxValue);
+
+        populateIcons();
+    }
+
+    private int getMaxScrollOffset() {
+        int maxItemOffset = Math.max(0, usedIcons.size() - maxIconsToDisplay);
+        return maxItemOffset / iconsPerRow + (maxItemOffset % iconsPerRow > 0 ? 1 : 0);
+    }
+
+    private void populateIcons() {
+        for (AbstractWidget widget : iconFilterWidgets) {
+            this.removeWidget(widget);
+        }
+
+        this.iconFilterWidgets.clear();
+
+        // Starting Y position for the icons
+        int row = (int) ((int) (dividedHeight * HEADER_HEIGHT) + (dividedHeight / 2f));
+        // Starting X position for the row
+        int xPos = (int) (dividedWidth * 13);
+
+        int currentIcon;
+
+        int widgetsOnCurrentRow = 0;
+
+        // Render icon widgets
+        for (int i = 0; i < maxIconsToDisplay; i++) {
+            // Get the icon to render
+            currentIcon = i + (scrollOffset * iconsPerRow);
+
+            // If there are less icons than MAX_ICONS_TO_DISPLAY, make sure we don't try and get a icon out of range
+            if (currentIcon > usedIcons.size() - 1) {
+                break;
+            }
+
+            Texture icon = usedIcons.get(currentIcon);
+
+            IconFilterWidget filterWidget = new IconFilterWidget(
+                    xPos, row, iconButtonSize, iconButtonSize, icon, this, icons.getOrDefault(icon, false));
+
+            iconFilterWidgets.add(filterWidget);
+
+            this.addRenderableWidget(filterWidget);
+
+            widgetsOnCurrentRow++;
+
+            // Calculate if we can place another icon button on the same row or not
+            if (widgetsOnCurrentRow == iconsPerRow) {
+                row += iconButtonSize;
+                xPos = (int) (dividedWidth * 13);
+                widgetsOnCurrentRow = 0;
+            } else {
+                xPos += iconButtonSize;
+            }
+        }
+    }
+
+    private void calculateIconSize() {
+        iconButtonSize = (int) (dividedWidth * DEFAULT_WIDGET_SIZE_MULTIPLIER);
+        iconsPerRow = MAX_ICONS_PER_ROW;
+        maxIconsToDisplay = MAX_ICONS_PER_PAGE;
+
+        // If there are less icons than the default per page, see how big we can make the icons whilst still staying on
+        // one page
+        if (usedIcons.size() < MAX_ICONS_PER_PAGE) {
+            int renderAreaWidth = (int) (dividedWidth * 36);
+            int renderAreaHeight = (int) (dividedHeight * 40);
+            int sizeToCheck;
+
+            // Maximum size should be fitting 2 icons on screen at once which will be roughly half of the width of the
+            // renderable area
+            for (int i = (DEFAULT_WIDGET_SIZE_MULTIPLIER + 1); i < MAX_WIDGET_SIZE_MULTIPLIER; i++) {
```

Comment:
I think keep the for loop's i variable clean, and do the size multiplication inside the loop